src/wasm/wasm-type.cpp

```diff
@@ -542,9 +542,16 @@ Type Type::getLeastUpperBound(Type a, Type b) {
     return Type::none; // a poison value that must not be consumed
   }
   if (a.isRef()) {
-    // FIXME: `anyref` is only valid here if the `anyref` feature is enabled,
-    // but this information is not available within `Type` alone.
-    return b.isRef() ? Type::anyref : Type::none;
+    if (b.isRef()) {
+      // Just returning `anyref` assumes that this method is never called with
+      // let's say `externref` and `funcref` where the anyref feature is not
+      // enabled, and that it would be a bug either in a pass or on the user's
+      // end during code generation if it did happen. If it turns out that it
+      // can happen otherwise, this method would need to take the enabled
+      // feature set as an additional argument at all call sites.
```

Comment:
- I think if user's input code was verified and accepted, this cannot be a problem in the user's end. If user's code had a problem, it should've been rejected when reading because it was using not enabled features.

> If it turns out that it can happen otherwise, this method would need to take the enabled feature set as an additional argument at all call sites.
- I don't think it can happen otherwise, but maybe it can, and I'm missing something. But if that happens we can fix it; I'm not sure we need to express uncertainties in the comments..
- I think feature set implication (so that if you enable only EH, reference types is also enabled) is something that's good to have regardless of whether this can happen or not. I was skeptical, but apparently other tools and VMs are doing it too. Of course that will be a future task (for anyone interested) and I'm not suggesting we should implement that here.

Reply:
Oh, with "the user's end" I meant a generator like AssemblyScript, using the C or JS APIs. A generator can freely emit `anyref`, or `select`s with differing reftype arms even if a feature is not enabled, which would be a bug on their end.

In general I still feel like this should be fixed properly instead, and I'm having a hard time coming up with a comment here that doesn't sound like "lol, this will most likely break". May I ask for a concrete suggestion I can copy and paste?

Reply:
```
The LUB of two different reference types is anyref, which may or may not be a valid type depending on whether the `anyref` feature is enabled. When `anyref` is disabled, it is possible for the finalization of invalid code to introduce a use of anyref via this function, but that is not a problem because it will be caught and rejected by validation.
```